wgpu-rs/src/lib.rs

```diff
@@ -170,10 +171,15 @@ impl Device {
     }
 
     pub fn create_pipeline_layout(&self, desc: &PipelineLayoutDescriptor) -> PipelineLayout {
+        //TODO: avoid allocation here
```

Comment:
If these structs became transparent new types around ID then I guess we wouldn't need these maps at all (unless we need to add more fields to these structs in the future)